src/common/services/api/donations.service.spec.js

```diff
@@ -65,4 +65,15 @@ describe( 'donations service', () => {
       $httpBackend.flush();
     } );
   } );
+
+  describe( 'getNextGiftDate( data )', () => {
+    it('should give next gift date', () => {
+      let data = {};
+      expect(donationsService.getNextGiftDate(data)).toBe(false);
+      data['recurring-day-of-month'] = '25';
+      // im not sure how to test this function without duplicating its functionality into here =))
```

Comment:
I'd use jasmine to mock the date and then test a couple of inputs and outputs. If current date (mocked by jasmine) is 8/10/2015 and recurring day of month is 9, I expect the output to be 9/9/2015. I'd just do a normal case and then one where it has to jump to the next month. You can do more if you want.